sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/models/AzureJsonHelper.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.monitor.opentelemetry.exporter.implementation.models;
+
+import com.azure.json.JsonWriter;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class AzureJsonHelper<T> {
+
+    // this method is replacing JsonWriter.writeMapField(), which would generate json attributes with null value.
+    // This method is used to write a map to a json writer.
+    // The map is written as a json object with the key as the field name and the value as the field value.
+    // The value is written using the writeValueCallback.
+    public static <T> void writeMap(Map<String, T> map, String fieldName, JsonWriter jsonWriter, Consumer<T> writeValueCallback) throws IOException {
+        if (map == null) {
+            return;
+        }
+
+        jsonWriter.writeFieldName(fieldName);
+        jsonWriter.writeStartObject();
+        for (Map.Entry<String, T> entry : map.entrySet()) {
+            jsonWriter.writeFieldName(entry.getKey());
+            writeValueCallback.accept(entry.getValue());
+        }
```

Comment:
Given all usages of this helper method check for the value not being null, I think this should be updated to:

```java
for (Map.Entry<String, T> entry : map.entrySet()) {
    T value = entry.getValue();
    if (value == null) {
        continue;
    }

    jsonWriter.writeFieldName(entry.getKey());
    writeValueCallback.accept(entry.getValue());
}
```

The current design will run into an exception if the value is null and serialization is skipped where it will try to write two field names in a row which is invalid.